plugins/BEdita/Core/src/Model/Table/AsyncJobsTable.php

```diff
@@ -328,4 +332,26 @@ protected function findPriority(Query $query, array $options)
             ->find('pending')
             ->orderDesc($this->aliasField('priority'));
     }
+
+    /**
+     * Update field "results" by entity, success and message
+     *
+     * @param \BEdita\Core\Model\Entity\AsyncJob $entity The Job entity
+     * @param bool $success The success flag
+     * @param array $messages The messages
+     * @return void
+     */
+    public function updateResults(AsyncJob $entity, bool $success, array $messages = []): void
+    {
+        $this->getConnection()->transactional(function () use ($entity, $success, $messages) {
+            $results = (array)$entity->get('results');
+            $attempt = count($results) + 1;
+            $data = compact('messages');
+            $result = compact('data', 'success');
+            $result['attempt_number'] = $attempt;
+            $results[] = $result;
+            $entity->set('results', $results);
+            $this->saveOrFail($entity);
```

Comment:
```suggestion
            $this->saveOrFail($entity, ['atomic' => false]);
```
To avoid running a transaction within a transaction.